packages/react/toggle-button/src/ToggleButton.tsx

```diff
@@ -1,10 +1,199 @@
+      ...groupProps
+    } = props;
+
+    const [value, setValue] = useControlledState<string[]>({
+      prop: valueProp,
+      defaultProp: defaultValue || [],
+      onChange: onValueChange,
+      isEqual: stringArraysAreEqual,
+    });
+
+    function handleChange(buttonValue: string) {
+      if (!value || value.length < 1) {
+        return setValue([buttonValue]);
+      }
+
+      if (selectionIsRequired && value.length === 1 && value.includes(buttonValue)) {
+        return;
+      }
+
+      setValue(
+        value.includes(buttonValue)
+          ? value.filter((v) => v !== buttonValue)
+          : value.concat(buttonValue).sort()
+      );
+    }
+
+    const context: ToggleButtonGroupContextValue = {
+      setValue,
+      value,
+      handleChange,
+      rovingFocus,
+      name: GROUP_CONTEXT_NAME,
+    };
+
+    return (
+      <Comp {...getPartDataAttrObj(GROUP_NAME)} role="group" ref={forwardedRef} {...groupProps}>
+        <ToggleButtonGroupContext.Provider value={context}>
+          <ToggleButtonGroupInner rovingFocus={rovingFocus}>{children}</ToggleButtonGroupInner>
+        </ToggleButtonGroupContext.Provider>
+      </Comp>
+    );
+  }
+);
+
+function ToggleButtonGroupInner(props: { rovingFocus: boolean; children: React.ReactNode }) {
+  return props.rovingFocus ? (
+    <RovingFocusGroup loop>{props.children}</RovingFocusGroup>
+  ) : (
+    <React.Fragment>{props.children}</React.Fragment>
+  );
+}
+
+/* -------------------------------------------------------------------------------------------------
+ * ToggleButtonExclusive
+ * -----------------------------------------------------------------------------------------------*/
+
+const GROUP_EXC_NAME = 'ToggleButtonGroupExclusive';
+const GROUP_EXC_CONTEXT_NAME = 'ToggleButtonGroupExclusiveContext';
+
+type ToggleButtonGroupExclusiveOwnProps = ToggleButtonGroupSharedProps & {
+  /** The controlled value of the toggled button */
+  value?: string | null;
+  /** The uncontrolled value of the toggled button */
+  defaultValue?: string;
+  /** A function called when the value of the toggled buttons changes */
+  onValueChange?(value: string | null): void;
+};
+
+type ToggleButtonGroupExclusiveContextValue = {
+  value: string | null;
+  setValue:
+    | React.Dispatch<React.SetStateAction<string | null>>
+    | React.Dispatch<React.SetStateAction<string>>;
+  handleChange(value: string | null): void;
+  name: typeof GROUP_EXC_CONTEXT_NAME;
+  rovingFocus: boolean;
+};
+const ToggleButtonGroupExclusiveContext = React.createContext<ToggleButtonGroupExclusiveContextValue | null>(
+  null
+);
+ToggleButtonGroupExclusiveContext.displayName = GROUP_EXC_CONTEXT_NAME;
+
+const ToggleButtonGroupExclusive = forwardRefWithAs<
+  typeof GROUP_DEFAULT_TAG,
+  ToggleButtonGroupExclusiveOwnProps
+>((props, forwardedRef) => {
+  const {
+    as: Comp = GROUP_DEFAULT_TAG,
+    value: valueProp,
+    defaultValue,
+    onValueChange,
+    children,
+    selectionIsRequired = false,
```

Comment:
Can we rename this to `required` please? This looks like the equivalent of `isSelectionRequired` and we agreed we would just do `disabled` not `isDisabled` etc. 

Also, it will match the native naming and make it consistent with `RadioGroup` 😛

Reply:
Yeah I wasn't sure about that since it isn't a form field and doesn't have any correlation with the native attribute. Probably fine either way though.

Reply:
Personally I think whether its a form field or not is just an implementation detail but, in `Radio` from `RadioGroup` we render a hidden input alongside the button in case someone uses it in a form. I wonder if we need to do the same for `ToggleButton` ?

Reply:
I could see that for a single value. How would multiple values serialize? We'd do some kind of multi checkboxes with name under the hood?

Reply:
Yeah, in `Slider` for example I change the `name` they pass to `${name}[]`. I forgot to do this for `RadioGroup` actually, probably should! but yeah, this is the common way arrays are handled when passing values to server via form `POST`.

Reply:
What if we just axe the `required` prop altogether for groups? If a consumer wants to require a selection they can do so by controlling state, and they can implement their own required prop if they want by wrapping the primitive. They can determine how they want to deal with validation, serialization, etc., if that's important to them. I can see cases where I might go either way.

Reply:
Yeah I did mention that to both @colmtuite and @jjenzz too, others do it and it's less code for us to maintain and it's really just state management at this point (ie. when do you allow setting the state or not).

Reply:
> What if we just axe the `required` prop altogether for groups?

I agree that we don't need to manage state for them. But, I do still have this on `RadioGroup` because it passes it to each of the underlying `input`s in case it is used in a form. 

I see it like `<input required />`. When we do this, it doesn't manage state that prevents you from removing the value, it just signals validation requirements and I imagine the `ToggleGroup` should behave the same (with underlying inputs).

Reply:
Yeah, I have am honestly not sure how I'd expect this to work as a consumer. Buttons can carry values in a form so would we not pass the `value` prop onto the native button in this case and instead pass it to an underlying hidden input as you're suggesting? I'm hesitant to even mess with form behavior at all because I feel like 9/10 times if you're using toggle buttons in a form you should *probably* be using a checkbox or radio group instead, and I'd want to encourage good UI if we can.

Reply:
Just to close the loop on this, I removed this prop. The storybook examples for controlled variants show how to get this behavior.